sdk/openai/Azure.AI.OpenAI/src/Custom/Audio/AudioTranscriptionOptions.cs

```diff
@@ -55,6 +56,30 @@ public partial class AudioTranscriptionOptions
     /// </remarks>
     public string DeploymentName { get; set; }
 
+    // CUSTOM CODE NOTE:
+    // The specification-level array of strings for timestamp granularities (["word","segment"]) is replaced by a
+    // [Flags] enum.
+
+    /// <summary>
+    /// The bitwise-aggregated timestamp granularity types that should be requested on an audio transcription
+    /// operation.
+    /// </summary>
+    /// <remarks>
+    /// <para>Transcription timing information requires the <see cref="AudioTranscriptionFormat.Verbose"/> setting
+    /// for <see cref="ResponseFormat"/>.</para>
+    /// <para>If no explicit value is set for <see cref="TimestampGranularityFlags"/>, segment-level timing
+    /// information will be provided by default.</para>
+    /// <para>Multiple timestamp granularities may be requested via joining values with the single-pipe |
+    /// operator, with e.g. the following requesting both word- and segment-level timing:
+    /// <code>
+    /// TimestampGranularityFlags = <see cref="AudioTimestampGranularity.Word"/> | <see cref="AudioTimestampGranularity.Segment"/>
+    /// </code>
+    /// </para>
+    /// </remarks>
+    public AudioTimestampGranularity TimestampGranularityFlags { get; set; }
```

Comment:
Should this be nullable?

Reply:
I don't believe so; if it were, there'd be no semantic difference between "flags not specified at all" and "flags specified with no flags set." The application to the request is just done based on flags checks, so "no flags" is "no options" irrespective of whether it's "0" or a hypothetical "null."